metadata/tests/test_examples.py

```diff
@@ -1,3 +1,5 @@
+"""Test submission examples and workflows at the API, with an in-memory DB."""
```

Comment:
Ok, here's the meat of the PR. This adds a ``TestModerationScenarios`` class to evaluate how hold/sticky-hold activity in the classic system impacts the submission state in this project.